src/Microsoft.PowerShell.Commands.Management/commands/management/SetContentCommand.cs

```diff
@@ -24,11 +24,16 @@ public class SetContentCommand : WriteContentCommandBase
         /// </param>
         internal override void BeforeOpenStreams(string[] paths)
         {
-            if (paths == null || paths.Length == 0)
+            if (paths is null)
             {
                 throw PSTraceSource.NewArgumentNullException(nameof(paths));
             }
 
+            if (paths.Length == 0)
+            {
+                throw PSTraceSource.NewArgumentException(null, nameof(paths));
```

Comment:
`PSTraceSource.NewArgumentException` has an overload that takes a parameter name only, and we should use that one.

Reply:
We seems need AI 😸